aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -2517,25 +2525,40 @@ void compute_T18_scale_square(
   scalar_t theta
 ) {
   // Scale
-  const auto s = at::max(
-    at::zeros_like(norm),
-    at::ceil(at::log2(norm / theta))
-  ).unsqueeze(-1).unsqueeze(-1).to(at::kLong);
-  const auto pow2s = at::pow(2, s);
-  const auto a_scaled = a / pow2s;
+  const auto s = (at::floor(at::log2(norm / theta)) + 1).clamp(/*min=*/0);
```

Comment:
Mind adding a small description of the algorithm on top (the sorting and all that stuff)